CHANGES.rst

```diff
@@ -270,6 +270,11 @@ astropy.io.fits
 astropy.io.misc
 ^^^^^^^^^^^^^^^
 
+- Fix a bug that caused tables with columns of strings to not be written to HDF5 with Python 3.x
+  If the table columns have data in form of string then a new table is made
+  from the original table with columns of strings replaced by columns of bytes.
+  The new table can now be written in Hdf5 file.[#7024]
```

Comment:
This PR is actually two separate things, a bug fix (for writing) and a new feature (for reading).  @bsipocz - this is maybe tricky from the backport perspective?
```
Bug fix
- Fix a bug where writing to HDF5 failed for for tables with columns of
  unicode strings.  Now those columns are first encoded to UTF-8 and
  written as byte strings. [#7024]

New feature
- Add a new ``character_as_bytes`` keyword to the HDF5 Table reading
  function to control whether byte string columns in the HDF5 file
  are left as bytes or converted to unicode.  The default is to read
  as bytes (``character_as_bytes=True``). [#7024]
```

Reply:
Hmm, could it be split into 2 PRs? If not, is the bug significant enough to add a new feature to a bugfix release?